KLP module fails to build with latest mainline [5.8-rc]

The current mainline tree `5.8-rc` fails to build the kernel livepatching module,
  during the module build stage, with the error:
 ```
meminfo.o: changed function: meminfo_proc_show
make -C /root/objtool/linux M=/root/.kpatch/tmp/patch
make[1]: Entering directory '/root/objtool/linux'
  LDS     /root/.kpatch/tmp/patch/kpatch.lds
  AR      /root/.kpatch/tmp/patch/built-in.a
  CC [M]  /root/.kpatch/tmp/patch/patch-hook.o
  LD [M]  /root/.kpatch/tmp/patch/livepatch-meminfo.o
  MODPOST /root/.kpatch/tmp/patch/Module.symvers
/root/.kpatch/tmp/patch/.output.o.cmd: No such file or directory
make[2]: *** [scripts/Makefile.modpost:111: /root/.kpatch/tmp/patch/Module.symvers] Error 1
make[1]: *** [Makefile:1669: modules] Error 2
make[1]: Leaving directory '/root/objtool/linux'
make: *** [Makefile:22: livepatch-meminfo.ko] Error 2
```
 
the commit 70f30cfe5b89 ("modpost: use read_text_file() and get_line() for reading
text files") would fail any `.o` used in the building a module without `*.cmd` file.
The `*.cmd` file, holds the information like gcc command line used to build the
`.o`, source file details and dependencies list. This information I guess its used
by kbuild to trigger rebuilding of `.o` or `.o.*` files in case of the dependencies
are updated.
 
`kpatch/kmod/patch/Makefile`:
```
  ...
  $(KPATCH_NAME)-objs += patch-hook.o output.o
 
  all: $(KPATCH_NAME).ko
 
  $(KPATCH_NAME).ko:
          $(KPATCH_MAKE)
 
  patch-hook.o: patch-hook.c kpatch-patch-hook.c livepatch-patch-hook.c
          $(KPATCH_MAKE) patch-hook.o
```

`output.o` doesn't uses the kernel build system but is hand build using
`create-diff-object`, hence the `.output.o.cmd` is not generated. One way to
address this issue would be created an empty `.output.o.cmd`, so that the kernel
build system is happy and it technically has no dependency on the kernel to
trigger re-build over using make -i (ignore) flag to bypass any such issue.

```
  --- a/kpatch-build/kpatch-build
  +++ b/kpatch-build/kpatch-build
@@ -963,6 +963,8 @@ ld -r $KPATCH_LDFLAGS -o ../patch/tmp_output.o $(find . -name "*.o") 2>&1 | logg
  
 if [[ "$USE_KLP" -eq 1 ]]; then
        cp "$TEMPDIR"/patch/tmp_output.o "$TEMPDIR"/patch/output.o || die
+       # empty .cmd file is required to keep KBUILD happy
+       touch "$TEMPDIR"/patch/.output.o.cmd
 else
        # Add .kpatch.checksum for kpatch script
        md5sum ../patch/tmp_output.o | awk '{printf "%s\0", $1}' > checksum.tmp || die
```
Makefile magic is something I don't understand well, I guess there are better ways
to fix this issue.

--- a/kpatch-build/kpatch-build
+++ b/kpatch-build/kpatch-build
@@ -963,6 +963,8 @@ ld -r $KPATCH_LDFLAGS -o ../patch/tmp_output.o $(find . -name "*.o") 2>&1 | logg
 
 if [[ "$USE_KLP" -eq 1 ]]; then
 	cp "$TEMPDIR"/patch/tmp_output.o "$TEMPDIR"/patch/output.o || die
+	# Avoid MODPOST warning (pre-v5.8) and error (v5.8+) with an empty .cmd file
+	touch "$TEMPDIR"/patch/.output.o.cmd || die
 else
 	# Add .kpatch.checksum for kpatch script
 	md5sum ../patch/tmp_output.o | awk '{printf "%s\0", $1}' > checksum.tmp || die

